storage/watchable_store.go

```diff
@@ -295,6 +296,79 @@ func (s *watchableStore) syncWatchings() {
 	slowWatchingGauge.Set(float64(len(s.unsynced)))
 }
 
+// RangeAllUnsynced ranges on all unsynced watchings.
+func (s *watchableStore) RangeAllUnsynced() {
+	totalLimit := 0
+	minRev, maxRev := int64(math.MaxInt64), int64(math.MinInt64)
+	for w := range s.unsynced {
+		if w.cur > 0 && w.cur <= s.store.compactMainRev {
+			log.Printf("storage: %v", ErrCompacted)
+			delete(s.unsynced, w)
+			continue
+		}
+		if w.cur > s.store.currentRev.main {
+			log.Printf("storage: %v", ErrFutureRev)
+			delete(s.unsynced, w)
+			continue
+		}
+		totalLimit += cap(w.ch) - len(w.ch)
+		if minRev >= w.cur {
+			minRev = w.cur
+		}
+		if maxRev <= w.cur {
+			maxRev = w.cur
+		}
+	}
+
+	min, max := newRevBytes(), newRevBytes()
+	revToBytes(revision{main: minRev}, min)
+	revToBytes(revision{main: maxRev}, max)
```

Comment:
@xiang90 Actually I don't think we need these lines:

```
		// restore index
 		switch {
 		case isTombstone(key):
 			s.store.kvindex.Tombstone(kv.Key, rev)
 		default:
 			s.store.kvindex.Restore(kv.Key, revision{kv.CreateRevision, 0}, rev, kv.Version)
 		}
```
And the error occurs from calling `Restore` method, which I don't think we need in this case.

And do we have documentation on `Tombstone`? Actually it's hard to understand the lines 
after `UnsafeRange` (especially Restore and Tombstone parts).

It runs well without `Tombstone` and `Restore`.

Thanks,

I will keep working on this.